irohad/network/sync_gate.hpp

```diff
@@ -0,0 +1,24 @@
+//
+// Created by Takumi Yamashita on 2017/07/11.
+//
+
+#ifndef IROHA_NETWORK_SYNCGATE_HPP
+#define IROHA_NETWORK_SYNCGATE_HPP
+namespace iroha {
+  namespace network {
+
+    /**
+     * Interface for downloading blocks from a network
+     */
+    class SyncGate {
```

Comment:
I don't think SyncGate is a descriptive name. Just looking at the name, it is not clear if this is for peer syncing or client syncing, or syncing blocks or syncing other types of information.

If this is only for downloading blocks and not syncing, then **BlockDownloader** is better (I don't like the name BlockLoader because the name could confuse people who miss the *network* in the path and they might think it is for loading blocks from the local DB).